pandas/core/algorithms.py

```diff
@@ -104,6 +104,12 @@ def _ensure_data(values, dtype=None):
             dtype = values.dtype
         else:
             # Datetime
+            if values.ndim > 1:
+                # Avoid calling the DatetimeIndex constructor as it is 1D only
+                asi8 = values.view('i8')
```

Comment:
I believe that ensure_data should only take 1d input at all times. Is there a case where it does not? (nb we shoul dprob document / type this)

Reply:
Yes, this gets called with 2D values from `DataFrame.rank`.

Reply:
Do you know if it is only rank? Because if so, it might be useful to add that as a comment for somebody later reading the code and wondering the same question where 2D things are passed to this.